docs/proposals/caching.adoc

```diff
@@ -0,0 +1,105 @@
+While workspace caching can provide a large benefit for some languages such as 
+Python and Node it also introduces complexities and could hide dependency issues. 
+As a consequence this proposal only focuses on a global cache. 
+A workspace caching proposal could be  made at a later time.
+
+== Background
+
+Currently there is a single PVC per project which is shared by all build pipelines. 
+As a consequence only a single build at a time should run for a project. At the moment due to #394 parallel builds can happen, but with #407 and #160 solved, it will be "one build at a time for a repository".
+
+The PVC is mounted as the workspace in all build tasks. 
+
+The `ods-start` task wipes the PVC at the beginning of each build so that no data persists between builds.
+
+Each running tekton task mounts the workspace PVC which typically takes a noticeable time. As a consequence it makes sense to not introduce additional tasks.  
+
+== Solution
+
+In the initial implementation instead of a single PVC per project one PVC per repo will be used. 
+
+The same PVC is used for the workspace (where a repo is checked out in) and the global cache.
+
+The global cache enables build tasks to store cached files for at least the following purposes:
+
+1. Dependency caching. Languages that use dependencies as packaged artifacts in Nexus may not benefit much from caching dependencies. Unless there is a clear indication that there is a large performance win, languages should not use the global cache if they can use Nexus. A language that currently is not supported by Nexus is go. Another example where a large performance win could be expected are `pnpm` or `yarn` which can install dependencies by referencing the artifacts on the same file system in a very efficient way. In other words there is no need to unpack/copy the dependencies from a global file system cache.
+
+2. Cache prior build output to support build skipping (see proposal at https://github.com/opendevstack/ods-pipeline/pull/423). Usage of the global cache for this purpose should work analog to how this is done for dependency caching but it is not further elaborated on here as this is covered in the mentioned proposal.
+
+3. There may be other usages for example for the gradle wrapper that could perhaps benefit from a global cache.
+
+The following new parameter is introduced to build tasks supporting dependency caching:
+
+* `cache-deps` a boolean defaulting to `false`. Only if set to `true` the task will used dependency caching. 
+
+In addition to the build task parameter, build tasks also receives the following:
+
+* File `.ods/cache-deps-parent-dir` contains an absolute path without trailing '/' to an existing directory (no whitespace or newlines). If file `.ods/cache-deps-parent-dir` does not exits the task must not use dependency caching. This is used to switch off dependency caching dynamically for example with a special tag in a git commit message. 
```

Comment:
I would propose to start without disabling via commit message. Do we need this file then at all?

Reply:
Strictly speaking we wouldn't need it. However without it the caching locations would need to be hardcoded, which I would rather not have.